src/Analyzers/Core/Analyzers/UseConditionalExpression/ForReturn/UseConditionalExpressionForReturnHelpers.cs

```diff
@@ -62,29 +74,53 @@ internal static class UseConditionalExpressionForReturnHelpers
             falseStatement = UseConditionalExpressionHelpers.UnwrapSingleStatementBlock(falseStatement);
 
             // Both return-statements must be of the form "return value"
-            if (!(trueStatement is IReturnOperation trueReturnOp) ||
-                !(falseStatement is IReturnOperation falseReturnOp) ||
-                trueReturnOp.ReturnedValue == null ||
-                falseReturnOp.ReturnedValue == null)
+            if (!IsReturnExprOrThrow(trueStatement) ||
+                !IsReturnExprOrThrow(falseStatement))
             {
                 return false;
             }
 
-            if (trueReturnOp.Kind != falseReturnOp.Kind)
+            trueReturn = trueStatement as IReturnOperation;
+            falseReturn = falseStatement as IReturnOperation;
+            trueThrow = trueStatement as IThrowOperation;
+            falseThrow = falseStatement as IThrowOperation;
+
+            // Can't convert to `x ? throw ... : throw ...` as there's no best common type between the two (even when
```

Comment:
Hmm is it possible to somehow share code between this file and the previous helper? Only different seems to be between it being a return/assignment operation. Only if it does not lead to more convoluted shared code though...

Reply:
done :)